code/datums/ai_laws.dm

```diff
@@ -423,13 +423,15 @@
 		zeroth = null
 		zeroth_borg = null
 		return
-	else
-		if(owner && owner.mind && owner.mind.special_role)
-			return
-		else
-			zeroth = null
-			zeroth_borg = null
+	if(owner && owner.mind && owner.mind.special_role)
+		return
+	if (istype(owner, /mob/living/silicon/ai))
+		var/mob/living/silicon/ai/A=owner
+		if(A && A.deployed_shell && A.deployed_shell.mind && A.deployed_shell.mind.special_role)

```

Comment:
`if(A?.deployed_shell?.mind?.special_role)`

Reply:
This seems less easily readable than having double-ampersands.

Reply:
It's less readable and pretty sure it doesn't get called often enough for nested overhead to matter.

Reply:
Is using the question marks quicker? I thought it was just shorthand for the same process that I had used with double-ampersands.

Reply:
The integrated manual implies that, yes. Specifically that it'll poke A only once instead of 4 times (deployed_shell 3, mind 2 respectively).

Reply:
Actually I find it much more readable after you get used to it, and it's DRYer.
And performance doesn't really matter as this is very rarely called.

Reply:
Stylistic differences aren't enforced so unless there's benefit to it this seems nitpicky.

Reply:
It's just less verbose for free

Reply:
Well, if it's all the same then, performance wise, I'd prefer to stick with the chained "and"s. It's consistent with the rest of the file, and in my opinion it's much more obvious to the next casual code diver that winds up looking into AI law change code.

Reply:
```suggestion
		if(A?.deployed_shell?.mind?.special_role)
```
It is both faster and in the case of more complicated conditionals makes it more clear to read what's happening.